### PR TITLE
Flip the concatenation switch

### PR DESCRIPTION
**Pull request by timholy**
Completes JuliaLang/julia#8599. Part of JuliaLang/LinearAlgebra.jl#255.

### CODE REVIEW

File: base/abstractarray.jl [hunk 1]
```diff
@@ -12,42 +12,13 @@ typealias RangeIndex Union{Int, Range{Int}, UnitRange{Int}, Colon}
...
-        vcat(X...)
-    end
-else
-    function vect(X...)
-        T = promote_typeof(X...)
-        #T[ X[i] for i=1:length(X) ]
-        # TODO: this is currently much faster. should figure out why. not clear.
-        copy!(Array(T,length(X)), X)
-    end
+const _oldstyle_array_vcat_ = false  # TODO: deprecate/remove in 0.6
```
Comment (tkelman) [inline base/abstractarray.jl]: move it to `deprecated.jl` if it's really just an appendix for the sake of letting people check it?

Comment (timholy) [inline base/abstractarray.jl]: Maybe I was being overzealous, but I was thinking that we might need to wait until the next release cycle: the whole point of adding this must have been to allow people to check, and it seems unfair to immediately issue a deprecation warning the moment anyone actually gets benefit from having used it. It doesn't seem to hurt anything to wait until another release cycle before deprecating this binding.

Comment (tkelman) [inline base/abstractarray.jl]: I just mean moving the definition to deprecated.jl so we remember to delete it eventually

Comment (timholy) [inline base/abstractarray.jl]: I see. My biggest concern is whether someone will just nuke the old deprecations when starting on the next cycle. But I could move it and have a HUGE comment explaining my intentions.

**Pull request merged**